libs/@local/hashql/eval/src/graph/read/filter.rs

```diff
@@ -13,11 +16,30 @@ use hashql_hir::node::{
 
 use super::{
     CompilationError, FilterCompilerContext, GraphReadCompiler,
-    error::{BranchContext, branch_unsupported, qualified_variable_unsupported},
-    path::CompleteQueryPath,
-    sink::FilterSink,
+    error::qualified_variable_unsupported, path::CompleteQueryPath, sink::FilterSink,
 };
 
+/// Checks if a [`Node`] represents a boolean literal with the specified value.
+///
+/// This helper function is used to optimize conditional expressions when one
+/// branch is a literal boolean value.
+///
+/// # Examples
+///
+/// ```ignore
+/// // Returns true if the node represents `true`
+/// assert!(is_bool(true_node, true));
+/// // Returns false if the node represents `false` when checking for `true`
+/// assert!(!is_bool(false_node, true));
+/// ```
+#[expect(clippy::wildcard_enum_match_arm)]
+fn is_bool(node: Node<'_>, value: bool) -> bool {
```

Comment:
I think `is_bool_primitive` or `is_bool_literal` would be more unambiguous. Otherwise, it could be interpreted if `node` evaluates to `value`